scheduler/reconcile.go

```diff
@@ -275,19 +275,36 @@ func (a *allocReconciler) computeDeploymentUpdates(deploymentComplete bool) {
 //     is detected by the deploymentWatcher during it will enter a paused
 //     state. This flag tells Compute we're paused or pending, so we should
 //     not make placements on the deployment.
-func (a *allocReconciler) computeDeploymentPaused() {
+func (a *allocReconciler) computeDeploymentPausedAndFailed() {
 	if a.deployment != nil {
 		a.deploymentPaused = a.deployment.Status == structs.DeploymentStatusPaused ||
 			a.deployment.Status == structs.DeploymentStatusPending
 		a.deploymentFailed = a.deployment.Status == structs.DeploymentStatusFailed
 	}
-	if a.deployment == nil {
-		if a.job.IsMultiregion() &&
-			a.job.UsesDeployments() &&
-			!(a.job.IsPeriodic() || a.job.IsParameterized()) {
 
-			a.deploymentPaused = true
-		}
+	// compute MRD jobs
+	a.computeMRDDeploymentPaused()
+}
+
+// computeMRDDeploymentPaused sets deployment options that are specific to
+// multi-region jobs
+func (a *allocReconciler) computeMRDDeploymentPaused() {
+	// consider only jobs without current deployment
+	if a.deployment != nil {
+		return
+	}
+
+	// consider only jobs that are MRD and that use deployments
+	if !(a.job.IsMultiregion() && a.job.UsesDeployments() && !(a.job.IsPeriodic() || a.job.IsParameterized())) {
+		return
+	}
```

Comment:
I'd break this up even further:

```suggestion
	// consider only jobs that are MRD 
	if !a.job.IsMultiregion() {
		return
	}	
	// consider only jobs that use deployments
	if !a.job.UsesDeployments() || (a.job.IsPeriodic() || a.job.IsParameterized()) {
		return
	}
```

Because returning early here reverses the sense of the original set of conditionals, it might also be worth considering if the `IsPeriodic`/`IsParameterized` even makes sense to include here now, as it should always be false if `!a.job.UsesDeployments`?

Reply:
You're right of course, I will remove superfluous checks, thanks. (this is what happens when you have long conditions!)